apps/api/src/app/user/ee.user.controller.ts

```diff
@@ -0,0 +1,30 @@
+import { Body, ClassSerializerInterceptor, Controller, UseGuards, UseInterceptors, Logger, Post } from '@nestjs/common';
+import { UserSessionData } from '@novu/shared';
+import { ApiExcludeController, ApiOperation, ApiTags } from '@nestjs/swagger';
+import { UserAuthGuard, UserSession } from '@novu/application-generic';
+import { UserEntity } from '@novu/dal';
+import { SyncExternalUserCommand } from './usecases/sync-external-user/sync-external-user.command';
+import { SyncExternalUser } from './usecases/sync-external-user/sync-external-user.usecase';
+
+@Controller('/users')
+@ApiTags('Users')
+@UseInterceptors(ClassSerializerInterceptor)
+@UseGuards(UserAuthGuard)
+@ApiExcludeController()
+export class EEUserController {
+  constructor(private readonly syncExternalUserUsecase: SyncExternalUser) {}
+
+  @Post('/')
+  @ApiOperation({
+    summary: 'Sync external Clerk user with internal db',
+  })
+  async syncExternalUser(@UserSession() user: UserSessionData): Promise<UserEntity> {
+    Logger.verbose('Syncing external Clerk user', user._id);
+
+    return this.syncExternalUserUsecase.execute(
+      SyncExternalUserCommand.create({
+        userId: user._id,
+      })
+    );
+  }
```

Comment:
❓ **question:** I guess we want to leave the old functionality for backward compatibility until we have the Web changes implemented? 
Just asking because I'm not sure how you decided to deal with the required changes in the Web app.

Reply:
This got replaced by `webhooks-clerk.controller.ts` 

All these Clerk changes will live under a standalone feature flag, therefore everything including web should work as it was until the flag is removed.
When `IS_CLERK_ENABLED = false` the original user controller will be used